text/0000-context-service.md

```diff
@@ -0,0 +1,200 @@
+
+This API must be experimental for the MVP:
+
+```js
+import { setContextExperimental } from 'lwc';
+import { XFooContext } from 'x/fooContext';
+
+setContextExperimental(elm, XFooContext, {
+    x: 1,
+    y: 2,
+});
+```
+
+Where the first argument is the element that provides the new context value into its flatten DOM subtree, the second argument is the defined context reference returned from `createContextExperimental` function call, and the 3rd argument is the _context value_ to be provided.
+
+Invariants:
+
+* `setContextExperimental()` must be called at least once before the insertion into the DOM phase is completed.
+
+Observations:
+* LWC is not opinionated on how a context definition is going to be shared. User-land abstractions are allowed.
+* LWC is not opinionated on how a developer will share the capabilities to set a new context value on an element. User-land abstractions are allowed.
+
+## Other examples
+
+### How to implement a React-like context HOC in LWC?
+
+```js
+import { LightningElement, createContextExperimental, setContextExperimental } from 'lwc';
+
+// creating a new context definition
+const Context = createContextExperimental({
+    x: 0,
+    y: 0,
+});
+
+export default class XFooContext extends LightningElement {
+    _value = {};
+
+    @api
+    get theme() {
+        return this._value;
+    }
+    set theme(v) {
+        // you can validate `v` here if you want
+        this._value = v;
+        // setting a new context value 
+        this.setNewContextValue();
+    }
+
+    connectedCallback() {
+        // setting the initial value of the context
+        this.setNewContextValue();
+    }
+    setNewContextValue() {
+        setContextExperimental(this, Context, this._value);
+    }
+    static Consumer = Context;
+}
+```
+
+The way a provider is used in a template is very straight forward:
+
+```html
+<template>
+    <x-foo-context theme="dark">
+        <c-consumer></c-consumer> 
+    </x-foo-context>
+    <x-foo-context value="lite">
+        <c-other-consumer></c-other-consumer> 
+    </x-foo-context>    
+</template>
+```
+
+Where `<c-consumer>` and `<c-other-consumer>` or any of their descendent will be able to consume the context value by using the `@wire(XFooContext.Consumer)` decorator, e.g.:
+
+```js
+import { LightningElement } from 'lwc';
+import XFooContext from 'x/fooContext';
+
+export default class MyComponent extends LightningElement {
+    @wire(XFooContext.Consumer) foo; // context object provided by <x-foo-context> element
+}
+```
+
+# Other Alternatives
+
+* The two new APIs can be imported from `@lwc/context` or `@lwc/wire-service` rather than `lwc` directly. This might help to eventually allow others to use contexts with vanilla web components. It will also make it easier for Lightning Platform to restrict who has access to these APIs in platform.
```

Comment:
```suggestion
* The two new APIs can be imported from `@lwc/context` or `@lwc/wire-service` rather than `lwc` directly. This might help to eventually allow others to use contexts with vanilla web components. It will also make it easier for Lightning Platform to restrict who has access to these APIs on platform.
```